### PR TITLE
issue #67 - overflowing the timer correctly according to int32_t

### PR DESCRIPTION
**Pull request by elad-nach**
i did not change the logic for handling `setTimeout` with a value which is smaller than 1. I think this is redundant since some users might be using `setTimeout` with a value of 0, then `tick()`ing it might result in an unexpected sequence of timers being executed. There's one test that demonstrates this (`"calls the given callback before setTimeout"`).
not sure if we need to change this edge case.

### CODE REVIEW

**Comment by benjamingr** [comment]
Mirroring my comment from before - I think we need to align with https://heycam.github.io/webidl/#abstract-opdef-converttoint for the timer.

File: src/lolex-src.js [hunk 1]
```diff
@@ -483,6 +484,7 @@ function createClock(now, loopLimit) {
         });
     };
     clock.setInterval = function setInterval(func, timeout) {
+        timeout = timeout > maxTimeout ? 1 : timeout;
```
Comment (benjamingr) [inline src/lolex-src.js]: I'd rather have all the bounds checking logic once in `addTimer`

File: test/lolex-test.js [hunk 2]
```diff
@@ -53,6 +53,28 @@ describe("issue #73", function () {
     });
 });
 
+describe("issue #67", function () {
+    // see https://nodejs.org/api/timers.html
+    it("should overflow to 1 on very big timeouts", function () {
```
Comment (benjamingr) [inline test/lolex-test.js]: This is good, I would prefer more tests :)

**Comment by elad-nach** [comment]
as I mentioned in the issue comment - I understand the IDL reference as a reference but I'm not sure about your request - are you asking to write the integer conversion as specified in the IDL?

**Comment by benjamingr** [comment]
@elad-nach I'm wondering if the correct thing would be to implement the IDL (which is what's discussed in the original issue) or not. I can't get Chrome to act according to the spec exactly - so I'm a little torn on what behavior we want.

**Comment by benjamingr** [comment]
In any way, I think we should land this - as it's better to specify this behavior in a useful way and to debate correctness after :)
